### ISSUE TITLE
What kernel version do I need to use bind_phc

### ISSUE THREAD

**Issue by teodly** [pendulum-project/timestamped-socket#66]
I'm using Linux kernel 6.9.9-100.fc39.x86_64, yet [socket.so_timestamping](https://github.com/pendulum-project/timestamped-socket/blob/ef24b9adccc4e9d565021e7ecbe3430c172deb97/src/socket/linux.rs#L147) returns EINVAL when [configure_timestamping](https://github.com/pendulum-project/timestamped-socket/blob/ef24b9adccc4e9d565021e7ecbe3430c172deb97/src/socket/linux.rs#L110) is called with Some bind_phc.

SOF_TIMESTAMPING_BIND_PHC is not documented as of Linux 6.10 (with /latest/ in the documentation URL): https://www.kernel.org/doc/html/latest/networking/timestamping.html .

```
# ethtool --version
ethtool version 6.9
# ethtool -T enpxxx
Time stamping parameters for enpxxx:
Capabilities:
        hardware-transmit
        software-transmit
        hardware-receive
        software-receive
        software-system-clock
        hardware-raw-clock
PTP Hardware Clock: 0
Hardware Transmit Timestamp Modes:
        off
        on
Hardware Receive Filter Modes:
        none
        all
        ptpv1-l4-sync
        ptpv1-l4-delay-req
        ptpv2-l4-sync
        ptpv2-l4-delay-req
        ptpv2-l2-sync
        ptpv2-l2-delay-req
        ptpv2-event
        ptpv2-sync
        ptpv2-delay-req
```

Capabilities don't include bind-phc, [should they?](https://github.com/torvalds/linux/blob/c0ecd6388360d930440cc5554026818895199923/net/ethtool/common.c#L430)

**Comment by davidv1992** [comment]
What configuration options were used to compile the kernel?

**Comment by teodly** [comment]
It is stock Fedora kernel.

[config-6.9.9-100.fc39.x86_64](https://github.com/user-attachments/files/16625052/config-6.9.9-100.fc39.x86_64.txt)

**Comment by davidv1992** [comment]
Dove into this, it is a bug on our end. bind_phc can only be done with vclocks, not with the native hardware clocks. Gonna build a fix.

**Comment by davidv1992** [comment]
Fixed in #70
